pruner/finder.go

```diff
@@ -0,0 +1,82 @@
+package pruner
+
+import (
+	"context"
+	"time"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+// maxHeadersPerLoop is the maximum number of headers to fetch
+// for a prune loop (prevents fetching too many headers at a
+// time for nodes that have a large number of pruneable headers).
+var maxHeadersPerLoop = uint64(1024)
+
+// findPruneableHeaders returns all headers that are eligible for pruning
+// (outside the sampling window).
+func (s *Service) findPruneableHeaders(ctx context.Context) ([]*header.ExtendedHeader, error) {
```

Comment:
@walldiss I understand the concern which is why there is a limit `maxHeadersPerLoop` which is 1024 headers that it's able to load into mem at a time.

Max header size rn (for 2MB square + 100 val set) is ~ 64KB so that would be something like 64MB which is large yes.

We can decrease the size of `maxHeadersPerLoop` to 512, wdyt?

Reply:
If it is a difficult fix right now, lets land it into the hardening issue and settle on lower `maxHeadersPerLoop`. Once the header size grows it will increase memory footprint again, so we would need to improve it eventually.

Reply:
How is 512? I think that's reasonable.

Reply:
FWIW our max range request size on the exchange level is 512 as well, meaning 512 headers are in mem at any given point during sync process.